### ISSUE TITLE
xfi fails to find files with 'find: -printf: unknown option' under OpenBSD

### ISSUE THREAD

**Issue by morgant** [morgant/xwf#10]
`xfi` fails to find files when clicking the 'Find' button with the following error:

```
find: -printf: unknown option
````

This is under OpenBSD 6.9-stable, which the `find` command in base does not, in fact, have a `-printf` option.

**Comment by morgant** [comment]
Removing the `-printf` portion of the call to `find` does work, but it appears that data sent through the pipe is not flushed upon each new line, so the results just appear all of the sudden when the `find` operation completes.

**Comment by morgant** [comment]
We might be able to use `openpty()` to make `xfi` execute `find` using a pseudo-terminal so that it flushes its output after every newline, but I don't think we want to add this complexity at this time. It might be better to make the [GtkProgressBar](https://refspecs.linuxbase.org/gtk/2.6/gtk/GtkProgressBar.html) indeterminate instead so that it is at least updating while `find` is being executed, but to do that we would need to implement threading, which definitely adds complexity.

**Comment by morgant** [comment]
I've removed the Linux-specific `-printf \"%%p\\n\\c\"` argument from `find` commands. This would have caused the `find` output to be nicely flushed after every matching file/directory, but is not portable. So, as a consequence, the UI (esp. the progress bar) doesn't update while `find` is executing. Oh, well.

I also removed some deprecated progress bar function calls and improved the status & progress bar updates, while I was in there.
